addons/website/controllers/main.py

```diff
@@ -45,14 +44,14 @@ def index(self, **kw):
         return self.page(page)
 
     #------------------------------------------------------
-    # Login - overwrite of the web login so that regular users are redirected to the backend 
+    # Login - overwrite of the web login so that regular users are redirected to the backend
     # while portal users are redirected to the frontend by default
     #------------------------------------------------------
     @http.route(website=True, auth="public")
     def web_login(self, redirect=None, *args, **kw):
         r = super(Website, self).web_login(redirect=redirect, *args, **kw)
         if not redirect and request.params['login_success']:
-            if request.registry['res.users'].has_group(request.cr, request.uid, 'base.group_user'):
+            if request.env.user.has_group('base.group_user'):
```

Comment:
Be carefull ! The `has_group` is a `api.model` method ; there you are checking if the user that browse `request.env.user` has the group, but the user from the environment is browse as sudo.
The good code is `request.env['res.users'].has_group('base.group_portal')`